src/sorcha_community_utils/activity/lsst_comet/lsst_comet_activity.py

```diff
@@ -0,0 +1,93 @@
+from sorcha.activity.base_activity import AbstractCometaryActivity
+from .model import Comet
+from typing import List
+import pandas as pd
+import numpy as np
+
+
+class LSSTCometActivity(AbstractCometaryActivity):
+    """
+    Calculating the coma magnitude of a comet with input specifications.
+
+    The observation dataframe provided to the ``compute``
+    method should have the following columns:
+
+    * ``H_r`` - TODO
+    * ``k`` - Dust falling exponential value (dust falling at rh^k)
+    * ``afrho1`` - Quantity of A'Hearn et al. (1984). at perihelion (cm). See notes.
+    * ``optFilter`` - TODO
+    * ``TrailedSourceMag`` - TODO
+
+    Notes:
+    ``afrho1`` - The product of albedo, filling factor of grains within the observer field
+    of view, and the linear radius of the field of view at the comet
+
+    This class is derived from ``lsstcomet`` by Mike Kelley (C)  LSST Solar System Scientific Collaboration 2019
+    """
+
+    def __init__(
+        self, required_column_names: List[str] = ["H_r", "k", "afrho1", "optFilter", "TrailedSourceMag"]
+    ) -> None:
+        super().__init__(required_column_names)
+
+    def compute(
+        self,
+        df: pd.DataFrame,
+        observing_filters: List[str],
+        rho: List[float],
+        delta: List[float],
+        alpha: List[float],
+    ) -> pd.DataFrame:
+        """
+        Returns numpy array of 0's with shape equal to the input dataframe
+        time column.
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            The ``observations`` dataframe provided by ``Sorcha``.
+        observing_filters : List[str]
+            The photometric filters the observation is taken in (the filter
+            requested that the coma magnitude be calculated for)
+        rho : List[float]
+            Heliocentric distance [units TODO]
+        delta : List[float]
+            Distance to Earth [units TODO]
+        alpha : List[float]
+            Description [units TODO]
+
+        Returns
+        -------
+        pd.DataFrame
+            The original ``observations`` dataframe, with updated magnitude values
+            based on results of `lsstcomet` coma calculations.
+        """
+
+        self._validate_column_names(df)
+
+        com = Comet(R=df["H_r"], k=df.k, afrho1=df.afrho1)
```

Comment:
k and afrho1 are the special columns that need to be added to make this run. We do error checking in sorcha for all other columns. Should we have this check that  k and afrho1 columns exist and die gracefully dies and outputs a message to the error log?

Reply:
no. because how do we know what columns will be required in a user developed function. If this is to serve as an example it should be done here.

Reply:
Fair enough :)

Reply:
So I think that the way we can make this work for all user provided subclasses is that, the user will define the column names that are required as we have done here in the `__init__` method. Then, in the parent class (in Sorcha) implementation of self._validate_column_names(), we'll check for the existence of the columns, if they are missing we'll do two things: 

1. Raise an exception and log a message to the terminal
2. Write out the same error message to the the error log file

This change is tracked in https://github.com/dirac-institute/sorcha/issues/512

Reply:
I'll resolve this comment here since the work is tracked in an issue in the sorcha repo.

Reply:
sounds good. thanks